src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,191 @@ public with sharing class ADDR_Addresses_TEST {
+    static testMethod void testEmptyAddressCountryIsPopulatedFromCountryCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testEmptyAddressCountryIsPopulatedFromCountryCode') {
+            return;
+        }
+
+        Contact con = new Contact();
+        con.put('MailingCountryCode', null);
+        Address__c addressEmptyCountry = new Address__c(MailingCountry__c = null);
+
+        Test.startTest();
+
+        String countryCode = 'US';
+        String country = 'United States';
+
+        ADDR_Addresses_TDTM.populateAddressCountryFromCountryCode(con, 'Mailing', addressEmptyCountry);
+        System.assertEquals(null, addressEmptyCountry.MailingCountry__c);
+
+        con.put('MailingCountryCode', countryCode);
+        ADDR_Addresses_TDTM.populateAddressCountryFromCountryCode(con, 'Mailing', addressEmptyCountry);
+        System.assertEquals(country, addressEmptyCountry.MailingCountry__c);
+
+        Test.stopTest();
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update the Address' Mailing State when it is empty or does not match the source sObject's StateCode 
+        and State & Country Picklists are enabled.
+    verify:
+        Does not populate the Address' Mailing State if the source sObject's StateCode is empty
+        Populate the Address' Mailing State with the source sObject's StateCode value if the Mailing State is empty
+        Populate the Address' Mailing State with the source sObject's StateCode value if the Mailing State does not match the StateCode
+    **********************************************************************************************************/
+    static testMethod void testEmptyAddressStateIsPopulatedFromStateCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testEmptyAddressStateIsPopulatedFromStateCode') {
+            return;
+        }
+
+        Contact con = new Contact();
+        con.put('MailingStateCode', null);
+        Address__c addressEmptyState = new Address__c(MailingState__c = null);
+
+        Test.startTest();
+
+        String stateCode = 'CA';
+
+        ADDR_Addresses_TDTM.populateAddressStateFromStateCode(con, 'Mailing', addressEmptyState);
+        System.assertEquals(null, addressEmptyState.MailingState__c);
+
+        con.put('MailingStateCode', stateCode);
+        ADDR_Addresses_TDTM.populateAddressStateFromStateCode(con, 'Mailing', addressEmptyState);
+        System.assertEquals(stateCode, addressEmptyState.MailingState__c);
+
+        addressEmptyState.MailingState__c = 'NY';
+        ADDR_Addresses_TDTM.populateAddressStateFromStateCode(con, 'Mailing', addressEmptyState);
+        System.assertEquals('stateCode', addressEmptyState.MailingState__c);
```

Comment:
Typo, `stateCode` instead of `'stateCode'`.

Reply:
Good catch!